kornia/core/image.py

```diff
@@ -0,0 +1,280 @@
+        >>> img = Image(data, ImageColor.RGB)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+        >>> assert (img.data == data).all()
+        >>> out = (img + 2 * img - 3 * img.clone()).sum()
+        >>> assert isinstance(out, Image)
+
+        >>> # from a bached torch.tensor
+        >>> data = torch.rand(2, 3, 4, 5)  # BxCxHxW
+        >>> img = Image(data, ImageColor.RGB)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+        >>> assert img.is_batch
+        >>> r, g, b = img[:, 0], img[:, 1], img[:, 2]
+        >>> mean_ch = img.mean(dim=1, keepdim=True)
+        >>> mean_ch.shape
+        torch.Size([2, 1, 4, 5])
+
+        >>> # from a numpy array (like opencv)
+        >>> data = np.ones((4, 5, 3), dtype=np.uint8)  # HxWxC
+        >>> img = Image.from_numpy(data, color=ImageColor.BGR)
+        >>> assert img.channels == 3
+        >>> assert img.resolution == (4, 5)
+
+        >>> # direct conversion from/to dlpack
+        >>> data = torch.rand(3, 4, 5)
+        >>> dlpack = Image(data, ImageColor.RGB).to_dlpack()
+        >>> img = Image.from_dlpack(dlpack, ImageColor.RGB)
+
+        >>> # apply any kornia function
+        >>> data = torch.rand(3, 4, 5)
+        >>> img = Image(data, color=ImageColor.RGB)
+        >>> img_gray = img.apply(kornia.color.rgb_to_grayscale)
+        >>> assert isinstance(img_gray, Image)
+        >>> assert img_gray.color == ImageColor.GRAY
+
+        >>> # denormalize for visualization
+        >>> data = torch.rand(3, 4, 5)
+        >>> img = Image(data, color=ImageColor.RGB, mean=[0.,0.,0], std=[255.,255.,255.])
+        >>> img_denorm = img.denormalize()
+        >>> assert img_denorm.mean().item() > 1.0
+    """
+    # defaults
+    _meta: Dict[str, Any] = {}
+    _meta['color'] = ImageColor.RGB
+    _meta['mean'] = None
+    _meta['std'] = None
+
+    @staticmethod
+    def __new__(
+        cls,
+        data: Tensor,
+        color: ImageColor,
+        mean: Optional[List[float]] = None,
+        std: Optional[List[float]] = None,
+        *args,
+        **kwargs
+    ):
+        return Tensor._make_subclass(cls, data, *args, **kwargs)
+
+    def __init__(
+        self, data: Tensor, color: ImageColor, mean: Optional[List[float]] = None, std: Optional[List[float]] = None
+    ) -> None:
+        self._meta['color'] = color
+        self._meta['mean'] = mean
+        self._meta['std'] = std
+
+    @property
+    def is_batch(self) -> bool:
+        """Return whether the image data is a batch of images."""
+        return len(self.data.shape) > 3
+
+    @property
+    def channels(self) -> int:
+        """Return the number channels of the image."""
+        return self.data.shape[-3]
+
+    @property
+    def height(self) -> int:
+        """Return the image height (columns)."""
+        return self.data.shape[-2]
+
+    @property
+    def resolution(self) -> Tuple[int, int]:
```

Comment:
size could be better. resolution somehow reminds of ppi stuff

Reply:
we have `.size` by default since we subclass tensor. This property maps to me quite well specific image properties, during testing it's useful e.g. when you resize to check on the resulting image resolution.